disnake/abc.py

```diff
@@ -991,15 +991,24 @@ async def _clone_impl(
         base_attrs: Dict[str, Any],
         *,
         name: Optional[str] = None,
+        category: Optional[Snowflake] = MISSING,
         reason: Optional[str] = None,
     ) -> Self:
-        base_attrs["permission_overwrites"] = [x._asdict() for x in self._overwrites]
-        base_attrs["parent_id"] = self.category_id
+        # if the overwrites are MISSING defaults to the
+        # original permissions of the channel
+        if not base_attrs["permission_overwrites"]:
+            base_attrs["permission_overwrites"] = [x._asdict() for x in self._overwrites]
+        if category is not MISSING:
+            base_attrs["parent_id"] = category.id if category else None
+        else:
+            # if no category was given don't chhange the category
+            base_attrs["parent_id"] = self.category_id
         base_attrs["name"] = name or self.name
+        channel_type = base_attrs["type"]
```

Comment:
Right now this errors with a key error for everything except text/news channels.
```suggestion
        channel_type = base_attrs.get("type") or self.type.value
```